networkx/utils/misc.py

```diff
@@ -357,32 +430,50 @@ def create_py_random_state(random_state=None):
         if random.Random instance, return it.
         if None or the `random` package, return the global random number
         generator used by `random`.
-        if np.random package, return the global numpy random number
-        generator wrapped in a PythonRandomInterface class.
-        if np.random.RandomState or np.random.Generator instance, return it
-        wrapped in PythonRandomInterface
+        if an np.random.Generator instance, or the np.random package, or
+        the global numpy random number generator, then return it
+        wrapped in a PythonRandomViaNumpyBits class.
+        if a PythonRandomViaNumpyBits instance, return it
+
+        # Provided for backward bit-stream matching with legacy code
+        if a np.randomRandomState instance and not the global numpy default,
+        return it wrapped in PythonRandomInterface
         if a PythonRandomInterface instance, return it
-    """
-    import random
-
-    try:
-        import numpy as np
-
-        if random_state is np.random:
-            return PythonRandomInterface(np.random.mtrand._rand)
-        if isinstance(random_state, np.random.RandomState | np.random.Generator):
-            return PythonRandomInterface(random_state)
-        if isinstance(random_state, PythonRandomInterface):
-            return random_state
-    except ImportError:
-        pass
 
+    Note: Conversion from older PythonRandomInterface to PythonRandomViaNumpyBits
+    is handled here to allow users of Legacy `numpy.random.RandomState` to exactly
+    match the legacy values produced. We assume that if a user cares about legacy
+    values, they are using a np.RandomState instance that is not the numpy default.
+    The default instance has state reset for each Python session. The Generator
+    class does not guarantee to maintain bit stream across versions. We wrap any
+    RandomState instance other than the default with `PythonRandomInterface`.
```

Comment:
We wrap any RandomState instance other than the default with ...
All other ... are wrapped with

This is rather confusing!